eth/catalyst/api.go

```diff
@@ -756,3 +758,55 @@ func (api *ConsensusAPI) heartbeat() {
 func (api *ConsensusAPI) ExchangeCapabilities([]string) []string {
 	return caps
 }
+
+// GetPayloadBodiesV1 implements engine_getPayloadBodiesByHashV1 which allows for retrieval of a list
+// of block bodies by the engine api.
+func (api *ConsensusAPI) GetPayloadBodiesByHashV1(hashes []common.Hash) []*types.Body {
+	var bodies []*types.Body
+	for _, hash := range hashes {
+		block := api.eth.BlockChain().GetBlockByHash(hash)
+		if block == nil {
+			bodies = append(bodies, nil)
+			continue
+		}
+		body := block.Body()
+		if body == nil {
+			bodies = append(bodies, nil)
+			continue
+		}
+		// We only want to return the transactions, not uncles
+		if len(body.Uncles) != 0 {
+			body.Uncles = make([]*types.Header, 0)
+		}
+		bodies = append(bodies, body)
+	}
+	return bodies
+}
+
+// GetPayloadBodiesByRangeV1 implements engine_getPayloadBodiesByRangeV1 which allows for retrieval of a range
+// of block bodies by the engine api.
+func (api *ConsensusAPI) GetPayloadBodiesByRangeV1(start, count uint64) []*types.Body {
+	var bodies []*types.Body
+	if api.eth.BlockChain().CurrentBlock().NumberU64() < start {
+		// Return [] if the requested range is past our latest block
+		return bodies
+	}
```

Comment:
```suggestion
        var bodies []*types.Body
	if api.eth.BlockChain().CurrentBlock().NumberU64() < start {
		// Return [] if the requested range is past our latest block
		return 
	}
	bodies = make([]*types.Body, count)
```
If you do this, you don't have to append `nil` later on